src/factories/locationFactory.ts

```diff
@@ -36,6 +39,39 @@ export namespace LocationFactory {
             uri = Utilities.combineUriWithUriBase(physLocation.artifactLocation.uri, uriBase);
 
             if (uri) {
+                // Check for artifact content and create our artifact content URI.
+                // In order to have artifact content, the log must have a valid artifact index
+                // (represented by artifact location index) and the "contents" property of the artifact
+                // must be defined and have one of the "binary, text, or rendered" properties set.
+                if (physLocation.artifactLocation.index !== undefined) {
+                    const run: sarif.Run | undefined = sarifLog.runs[runInfo.runIndex];
+                    if (run && run.artifacts) {
```

Comment:
Optional chaining.

Reply:
I would love to except the type-script compiler doesn't appear to like if checks like this:

if (sarifLog.runs[runInfo.runIndex]?.artifacts && 
sarifLog.runs[runInfo.runIndex].artifacts[physLocation.artifactLocation.index])

 "resource": "/d:/Gabriel/Documents/Repos/sarif-vscode-extension/src/factories/locationFactory.ts",
 "owner": "typescript",
 "message": "Object is possibly 'undefined'.",
 "owner": "typescript",

Decided not to fight it.

---
In reply to: [411738990](https://github.com/microsoft/sarif-vscode-extension/pull/252#discussion_r411738990) [](ancestors = 411738990)